backend/api/Database/Models/Mission.cs

```diff
@@ -103,6 +104,27 @@ public static MissionStatus MissionStatusFromString(string status)
                   )
             };
         }
+
+        public void CalculateEstimatedDuration()
+        {
+            decimal robotVelocity = (decimal)1.5 * 1000 / 60; // km/t => m/min
+            decimal efficiencyFactor = (decimal)0.20;
+            decimal inspectionTime = 2; // min/tag
+            int assumedXYMetersFromFirst = 20;
+
+            double distance = 0;
+            int numberOfTags = 0;
+            var prevPosition = new Position(PlannedTasks.First().Pose.Position.X + assumedXYMetersFromFirst, PlannedTasks.First().Pose.Position.Y + assumedXYMetersFromFirst, PlannedTasks.First().Pose.Position.Z);
+            foreach (var plannedTask in PlannedTasks)
+            {
+                numberOfTags += plannedTask.Inspections.Count;
+                var currentPosition = plannedTask.Pose.Position;
+                distance += Math.Abs(currentPosition.X - prevPosition.X) + Math.Abs(currentPosition.Y - prevPosition.Y);
+                prevPosition = currentPosition;
+            }
+            int estimate = (int)(Divide((decimal)distance, robotVelocity * efficiencyFactor) + (numberOfTags * inspectionTime)); // min
+            EstimatedDuration = TimeSpan.FromMinutes(estimate);
+        }
```

Comment:
Make a test for this function?

I think perhaps all the 'magic numbers' used in this function can either be moved to a config file, or should in the future come from the robot type.
That way we can make a test for the function without having to keep ensuring that the test and this function has the same config values.

Reply:
Not sure it makes sense to test this function, since its an estimate, we can't really test the functionality other than the exact results it gives us now?

Reply:
Leaves the constants as is, since this is a temporary solution for the constants and they will be replaced with robot specific constants at a later time.